src/router.js

```diff
@@ -281,7 +281,7 @@ function validateRouteConfig(config) {
   }
 }
 
-function evaluateNavigationStrategy(instruction, evaluator, context){
+function evaluateNavigationStrategy(instruction:NavigationInstruction, evaluator, context): NavigationInstruction {
```

Comment:
I believe `evaluator` is a function, and `context` is a route config object (`Object` is fine for now).